src/agentlab/agents/generic_agent/generic_agent.py

```diff
@@ -260,3 +260,128 @@ def get_action_post_hoc(agent: GenericAgent, obs: dict, ans_dict: dict):
         output += f"\n<action>\n{action}\n</action>"
 
     return system_prompt, instruction_prompt, output
+
+
+def get_action_post_hoc(agent: GenericAgent, step_info):
+    """
```

Comment:
I think this can be safely ported to the unsupervised project

Reply:
we can remove this too. This is some old stuff i was trying